stubs/mypy-extensions/mypy_extensions.pyi

```diff
@@ -1,13 +1,19 @@
 import abc
+import sys
 from _typeshed import IdentityFunction, Self
 from collections.abc import ItemsView, KeysView, Mapping, ValuesView
-from typing import Any, Generic, TypeVar, overload
+from typing import Any, ClassVar, Generic, TypeVar, overload, type_check_only
 
 _T = TypeVar("_T")
 _U = TypeVar("_U")
 
 # Internal mypy fallback type for all typed dicts (does not exist at runtime)
+# N.B. Keep this mostly in sync with typing(_extensions)._TypedDict
+@type_check_only
 class _TypedDict(Mapping[str, object], metaclass=abc.ABCMeta):
+    __total__: ClassVar[bool]
+    # Unlike typing(_extensions).TypedDict,
+    # subclasses of mypy_extensions.TypedDict do NOT have the __required_keys__ and __optional_keys__ ClassVars
```

Comment:
@sobolevn, this is a good reason why we can't just re-export `typing_extensions._TypedDict` in `mypy_extensions` :(